devices/Hx711/Hx711.cs

```diff
@@ -0,0 +1,225 @@
+
+using System;
+using System.Device.Spi;
+using System.Diagnostics;
+using System.Threading;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// Scale class proving a wrapper to the Hx711 to be used as a scale for weight measurements.
+    /// </summary>
+    public class Scale
+    {
+        // number of clock periods to clock out in order to read a sample
+        private const int ClockPeriodsForSampleReading = 24;
+
+        // pulse train required to read a sample and setup gain factor for next reading
+        private byte[] readSamplePulseTrain;
+
+        private readonly bool[] _rawSample = new bool[24];
+        private readonly SpiDevice _spiDevice;
+
+        /// <summary>
+        /// Gets or sets the value that's subtracted from the actual reading.
+        /// </summary>
+        public int Offset { get; set; }
+
+        /// <summary>
+        /// Gets or sets the gain factor that the Hx711 uses when sampling.
+        /// </summary>
+        /// <remarks>
+        /// The default value is <see cref="GainLevel.Gain_128"/>.
+        /// </remarks>
+        public GainLevel Gain { get; set; }
+
+        /// <summary>
+        /// Gets or sets the number of samples that will be taken and then averaged when performing a <see cref="Read"/> operation.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The default value is 3 samples.
+        /// </para>
+        /// <para>
+        /// The number of samples to take has impact on the time it takes for a <see cref="Read"/> operation to complete. The Hx711, in the current configuration, as sample rate of 10 samples per second.
+        /// </para>
+        /// </remarks>
+        public uint SampleAveraging { get; set; } = 3;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Scale"/> class.
+        /// </summary>
+        /// <param name="spiDevice">The <see cref="SpiDevice"/> that is used as channel to communicate with the Hx711.</param>
+        /// <param name="gain"><see cref="GainLevel"/> that will be used for the scale. If not provided, the default is <see cref="GainLevel.Gain_128"/>.</param>
+        public Scale(
+            SpiDevice spiDevice,
+            GainLevel gain = GainLevel.Gain_128
+            )
+        {
+            _spiDevice = spiDevice;
+            Gain = gain;
+
+            // setup the pulse train required to read a sample AND setup gain factor for next reading
+            // gain factor from property
+            readSamplePulseTrain = new byte[]
+            {
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                0b1010_1010,
+                (byte)Gain
+            };
+        }
+
+        /// <summary>
+        /// Read weight from the scale.
+        /// </summary>
+        /// <returns></returns>
+        public double Read()
```

Comment:
What is the unit of the scale?
I guess it's grams or equivalent. In that case, with our pattern, better use the UnitsNet

Reply:
Thought about adding that, but left it as a raw value for the moment. Let's to that improvement in the next revision OK?

Reply:
I would rather do it now. So we are driving consistence across all the bindings

Reply:
I agree with you on the principle of consistency, just that we have a... _slight_ problem right now: there is no nanoFramework port for Mass on UnitsNet...